cpp/posix.h

```diff
@@ -80,12 +80,14 @@ inline int close(int fd) {
   return ::close(fd);
 }
 
-inline int putenv(Str* name, Str* value) {
-  int env_string_size =
-      name->len_ + value->len_ + 1;  // NOTE(Jesse): +1 for the '=' between them
-  char* env_string = static_cast<char*>(malloc(env_string_size));
-  snprintf(env_string, env_string_size, "%s=%s", name->data_, value->data_);
-  return ::putenv(env_string);
+inline void putenv(Str* name, Str* value) {
+  assert(name->IsNulTerminated());
+  assert(value->IsNulTerminated());
+  int overwrite = 1;
+  bool err = ::setenv(name->data_, value->data_, overwrite) == -1;
```

Comment:
The `bool` is confusing because often means success is true == 1.

-- I think we should use this C-style error handling in most places.  (We're not that consistent about it, e.g. in pipe() too)


```
int result = c_style_call()
if (result < 0) {
  throw new IOError(errno)
}
```